### PR TITLE
Fix: Refresh user status  on Patient Screen

### PR DESCRIPTION
**Pull request by sanchita**
https://www.pivotaltracker.com/story/show/167533992

### CODE REVIEW

File: app/src/test/java/org/simple/clinic/home/patients/PatientsScreenControllerTest.kt [hunk 1]
```diff
@@ -108,43 +107,26 @@ class PatientsScreenControllerTest {
...
     whenever(approvalStatusApprovedAt.get()).thenReturn(Instant.now())
     whenever(hasUserDismissedApprovedStatus.asObservable()).thenReturn(Observable.just(false))
     whenever(hasUserDismissedApprovedStatus.get()).thenReturn(false)
 
     uiEvents.onNext(ScreenCreated())
     uiEvents.onNext(TheActivityLifecycle.Resumed())
-    uiEvents.onNext(TheActivityLifecycle.Resumed())
 
-    verify(userSession, never()).refreshLoggedInUser()
+    verify(userSession, times(2)).refreshLoggedInUser()
```
Comment (vinaysshenoy) [inline app/src/test/java/org/simple/clinic/home/patients/PatientsScreenControllerTest.kt]: It's not obvious from this check that each of the two different `UiEvent`s (`ScreenCreated` and `Resumed`) can independently trigger a user refresh. Can we either:
1. Pass the event in via the test parameter as well so we can test each event separately, or
2. Make one verification call after the first event, and another after the second event so we know that each event triggers a refresh independently.

```kotlin
uiEvents.onNext(ScreenCreated())
verify(userSession).refreshLoggedInUser()

clearInvocations(userSession)

uiEvents.onNext(TheActivityLifecycle.Resumed())
verify(userSession).refreshLoggedInUser()
```

File: app/src/test/java/org/simple/clinic/home/patients/PatientsScreenControllerTest.kt [hunk 2]
```diff
@@ -177,6 +159,8 @@ class PatientsScreenControllerTest {
     whenever(userSession.loggedInUser()).thenReturn(Observable.just(Just(user)))
     whenever(hasUserDismissedApprovedStatus.asObservable()).thenReturn(Observable.just(false))
     whenever(hasUserDismissedApprovedStatus.get()).thenReturn(false)
+    whenever(userSession.refreshLoggedInUser()).thenReturn(Completable.never())
```
Comment (vinaysshenoy) [inline app/src/test/java/org/simple/clinic/home/patients/PatientsScreenControllerTest.kt]: We could move this to the `setUp` fixture and only override it in the tests that depend on this?

File: app/src/test/java/org/simple/clinic/home/patients/PatientsScreenControllerTest.kt (hunk 1 above)
Comment (sanchita) [reply app/src/test/java/org/simple/clinic/home/patients/PatientsScreenControllerTest.kt]: Good point.

File: app/src/main/java/org/simple/clinic/home/patients/PatientsScreenController.kt [hunk 3]
```diff
@@ -93,13 +93,14 @@ class PatientsScreenController @Inject constructor(
         // So duplicate triggers are dropped by restricting flatMap's
```
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/home/patients/PatientsScreenController.kt]: If you look at the `PatientsScreenController#refreshUserStatus` method, we can see that a complete data sync happens whenever the user gets refreshed if they are logged in.

This was meant to handle the case where sync should happen as soon as a user is approved. Now with this change, a complete sync will happen every time the screen is opened.

In order to maintain the current functionality, we should do this sync only whenever the user's logged in status and user status changes to (`Allowed`, Logged In`) from any other state.

Comment (sanchita) [reply app/src/main/java/org/simple/clinic/home/patients/PatientsScreenController.kt]: I will raise a new PR for this fix.

**Pull request merged**